sources/osg/Utils.js

```diff
@@ -226,5 +226,59 @@ define( [
 
     } )();
 
+    Utils.hashComputeCodeFromString = function ( str ) {
+
+        var hash = 0;
+        var chara = 0;
+
+        if ( str === undefined || str.length === 0 ) {
+            return hash;
+        }
+
+        for ( var i = 0; i < str.length; i++ ) {
+
+            chara = str.charCodeAt( i );
+            /*jshint bitwise: false */
+            hash = ( ( hash << 5 ) - hash ) + chara;
+            hash = hash & hash; // Convert to 32bit integer
+            /*jshint bitwise: true */
+
+        }
+
+        if ( hash < 0 ) {
+            hash = -hash;
+        }
+
+        return hash;
+
+    };
+
+    Utils.hashComputeCodeFromIntList = function ( intList ) {
+
+        var hash = 0;
+        var intAt = 0;
+
+        if ( intList === undefined || intList.length === 0 ) {
```

Comment:
Maybe ``if ( !intList || intList.length === 0 )`` because if it's null ``intList.length`` won't fail but....

@cedricpinson Can we agree on ``!obj`` vs ``obj === undefined`` when we test an **object**?
I often made this comment but I don't know if we agreed or not on this practice